8.0 349 fix month period

Roodin opened the pull request:
Según conversación en incidencia  #366
Se ha añadido también un pequeño fix en el l10n_es_aeat

Roodin committed:
[FIX]. l10n_es_aeat_mod349 . Recuperar funcionalidad perdida para poder calcular el 349 en períodos distintos de los fiscales

Roodin committed:
[FIX]. l10n_es_aeat_mod349 . Recuperar funcionalidad perdida para poder calcular el 349 en períodos distintos de los fiscales

Roodin committed:
[FIX].l10n_es_aeat. Correccion de Warning en onchange de period_type

Roodin committed:
[FIX].l10n_es_aeat. Correccion de Warning en onchange de period_type

Roodin committed:
l10n_es_aeat,  l10n_es_aeat_mod349. Pequeñas mejoras

Roodin committed:
l10n_es_aeat,  l10n_es_aeat_mod349. Pequeñas mejoras

pedrobaeza commented on l10n_es_aeat_mod349/views/mod349_view.xml:
```diff
@@ -188,6 +188,12 @@
                 <button name="%(l10n_es_aeat.action_wizard_aeat_export)d" position="attributes">
                     <attribute name="name">%(action_wizard_aeat_mod349_export)d</attribute>
                 </button>
+                <field name="period_type" position="replace">
```
Prefiero que no hagas un replace (ver luego charla sobre compatibilidad en las jornadas). Aunque quede un poco más "feo", ponlo a continuación.

pedrobaeza replied:
Ya, si es bastante improbable, pero utilizando el mismo criterio siempre no hay posibilidad de error. Puesto que tenemos también el control del "super", podemos poner:

```
<div name="period_type_container">
    <label for="period_type">
    <field name="period_type">
</div
```

Y ya meter dentro de ese div el check en este módulo.

Roodin replied:
¿Y no es  bastante lío tener que modificar esto en el aeat base y tener que
actualizarlo solo por este replace?. De verdad que comparto lo de no usar
replace pero creo que todo es evaluable y se deben ver pros y contras y
como dices es poco probable que dé problemas en este contexto.

El 3 de junio de 2016, 13:11, Pedro M. Baeza notifications@github.com
escribió:

> In l10n_es_aeat_mod349/views/mod349_view.xml
> https://github.com/OCA/l10n-spain/pull/367#discussion_r65691694:
> 
> > @@ -188,6 +188,12 @@
> >                  <button name="%(l10n_es_aeat.action_wizard_aeat_export)d" position="attributes">
> >                      <attribute name="name">%(action_wizard_aeat_mod349_export)d</attribute>
> >                  </button>
> > -                <field name="period_type" position="replace">
> 
> Ya, si es bastante improbable, pero utilizando el mismo criterio siempre
> no hay posibilidad de error. Puesto que tenemos también el control del
> "super", podemos poner:
> 
> <div name="period_type_container">
>     <label for="period_type">
>     <field name="period_type">
> </div
> 
> Y ya meter dentro de ese div el check en este módulo.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/OCA/l10n-spain/pull/367/files/df8b8a5d94757ae1a5a4332bf4a9860aec3feef7#r65691694,
> or mute the thread
> https://github.com/notifications/unsubscribe/AKpVCyCObu8QrgTL2Y4xZ5Lm9AniY_qgks5qIAvmgaJpZM4ItbTI
> .

## 

# 

Santi Argüeso
Twitter: @santiarg

Web: http://comunitea.com
Twitter: @comutuitea
Tfno: 982814805

pedrobaeza replied:
Bueno, lo dejo por esta vez por no bloquear esto mucho, pero sigo pensando que no queda tan feo teniendo el check en la siguiente línea haciendo un `position="after"`

Roodin replied:
Tks!!!
No es por feo, es por tener la relación entre los dos campos más clara.

2016-06-03 13:27 GMT+02:00 Pedro M. Baeza notifications@github.com:

> In l10n_es_aeat_mod349/views/mod349_view.xml
> https://github.com/OCA/l10n-spain/pull/367#discussion_r65693050:
> 
> > @@ -188,6 +188,12 @@
> >                  <button name="%(l10n_es_aeat.action_wizard_aeat_export)d" position="attributes">
> >                      <attribute name="name">%(action_wizard_aeat_mod349_export)d</attribute>
> >                  </button>
> > -                <field name="period_type" position="replace">
> 
> Bueno, lo dejo por esta vez por no bloquear esto mucho, pero sigo pensando
> que no queda tan feo teniendo el check en la siguiente línea haciendo un
> position="after"
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/OCA/l10n-spain/pull/367/files/df8b8a5d94757ae1a5a4332bf4a9860aec3feef7#r65693050,
> or mute the thread
> https://github.com/notifications/unsubscribe/AKpVCySTi-zmA7L2VwOajZ1tfqE_hQnMks5qIA-PgaJpZM4ItbTI
> .

## 

# 

Santi Argüeso
Twitter: @santiarg

Web: http://comunitea.com
Twitter: @comutuitea
Tfno: 982814805

Roodin commented:
Ok, me lo temía, lo sé, no creo que se vaya  a heredar desde aquí, pero acepto alternativas para dejarlo dentro del group

Roodin commented:
upss, te leí rápido, ¿no queda feo y podo "entendible" a continuación?

Roodin commented:
Comprendo lo del replace, ¿pero en este contexto no es preferible?

pedrobaeza commented on l10n_es_aeat_mod349/models/mod349.py:
```diff
@@ -117,6 +119,34 @@ def _create_349_refund_records(self, refunds):
+    def _get_domain(self):
+        self.ensure_one()
+        domain = [('state', 'in', ['open', 'paid']),
+                  ('period_id', 'in', self.periods.ids),
```
Deberías añadir el dominio de los periodos sólo si no es de tipo fecha.

Roodin replied:
mmm, no lo creo, prefiero tener los dos filtros, son compatible y el aviso
del checkbox ya lo digo así.

2016-06-03 13:12 GMT+02:00 Pedro M. Baeza notifications@github.com:

> In l10n_es_aeat_mod349/models/mod349.py
> https://github.com/OCA/l10n-spain/pull/367#discussion_r65691744:
> 
> > @@ -117,6 +119,34 @@ def _create_349_refund_records(self, refunds):
> >                       'amount_untaxed': refund.cc_amount_untaxed})
> >          return True
> > -    def _get_domain(self):
> > -        self.ensure_one()
> > -        domain = [('state', 'in', ['open', 'paid']),
> > -                  ('period_id', 'in', self.periods.ids),
> 
> Deberías añadir el dominio de los periodos sólo si no es de tipo fecha.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/OCA/l10n-spain/pull/367/files/df8b8a5d94757ae1a5a4332bf4a9860aec3feef7#r65691744,
> or mute the thread
> https://github.com/notifications/unsubscribe/AKpVC7d-DCBc4NQHVqwuHBqZOw2LhUBdks5qIAwMgaJpZM4ItbTI
> .

## 

# 

Santi Argüeso
Twitter: @santiarg

Web: http://comunitea.com
Twitter: @comutuitea
Tfno: 982814805

pedrobaeza replied:
OK

pedrobaeza commented on l10n_es_aeat_mod349/models/mod349.py:
```diff
@@ -216,6 +244,13 @@ def button_confirm(self):
+    calculate_date = fields.Boolean(
+        string='Calculate by days',
+        states={'confirmed': [('readonly',True)]},
+        help="Warning!: With this checkbox selected ,the declaration will "
```
Pon la coma bien

pedrobaeza commented on l10n_es_aeat_mod349/models/mod349.py:
```diff
@@ -216,6 +244,13 @@ def button_confirm(self):
     number = fields.Char(default='349')
+    calculate_date = fields.Boolean(
+        string='Calculate by days',
+        states={'confirmed': [('readonly',True)]},
```
Deja un espacio tras la coma

pedrobaeza commented:
Cambia las versiones de los módulos según https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#version-numbers

pedrobaeza commented:
@Roodin, mira la solución alternativa que te propongo.

pedrobaeza commented on l10n_es_aeat_mod349/models/mod349.py:
```diff
@@ -216,6 +244,13 @@ def button_confirm(self):
+        states={'confirmed': [('readonly',True)]},
+        help="Warning!: With this checkbox selected ,the declaration will "
+             "be computed selecting invoices based in periods and dates in "
+             "selected Period Type no only by period"
```
s/no/not

Roodin committed:
l10n_es_aeat. Pequeños retoques de código, según correcciones propuestas

Roodin committed:
l10n_es_aeat. Pequeños retoques de código, según correcciones propuestas

pedrobaeza committed:
[IMP] account_balance_reporting: Allow to not print certain accounts in level unfolding

pedrobaeza committed:
[FIX] account_balance_reporting: Show correct balance on intermediate accounts

pedrobaeza committed:
[FIX] account_balance_reporting: Make current_line_ids work

ORM is not handling correctly 2 many2many for the same comodel with default
relation, column1 and column2 values, because it links to the same table.
This fix assigns manually the name for the second field.

pedrobaeza committed:
[IMP] account_balance_reporting: Translation

pedrobaeza committed:
Merge pull request #368 from Tecnativa/8.0-account_balance_reporting-not_print-level_fix

8.0 account balance reporting not print level fix

OCA-git-bot committed:
[UPD] addons table in README.md

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[FIX] l10n_es_partner: Use requests library

pedrobaeza commented:
Queda por solucionar:

```
l10n_es_aeat_mod349/models/mod349.py:12:1: F401 'datetime' imported but unused
l10n_es_aeat_mod349/models/mod349.py:143:41: E128 continuation line under-indented for visual indent
l10n_es_aeat_mod349/models/mod349.py:146:60: E502 the backslash is redundant between brackets
```

y si haces un rebase, la rama debe salir en verde.

Roodin committed:
l10n_es_aeat_mod349. Fix PEP8

Roodin committed:
MERGE

Roodin committed:
l10n_es_aeat_mod349. Fix space PEP8 :(

pedrobaeza commented:
:+1:
